### PR TITLE
Prefill REDCap enrollment questionnaire from UW IdP attributes

### PR DESCRIPTION
**Pull request by tsibley**
Saves the participant time from entering answers already known to the
University and hopefully makes our data collection more complete and
standardized.

### CODE REVIEW

**Comment by kairstenfay** [comment]
[Populate uw_school in redcap via musher](https://trello.com/c/1sPyfvdA/810-populate-uwschool-in-redcap-via-musher)

**Review by kairstenfay — APPROVED** [review]
I'm not quite sure how to test this locally without the proper environment variables present for these changes, but they look good to me.

File: lib/husky_musher/app.py [hunk 1]
```diff
@@ -152,3 +156,77 @@ def main():
...
+    Extracts attributes of the authenticated user, provided by UW's IdP via our
+    Shibboleth SP, from the request environment *environ*.
+
+    Keys of the returned dict match those used by our REDCap project.
+    """
+    return {
+        # We ask in REDCap for their @uw.edu email address, so provide it in
+        # that same format.  On ingest from REDCap into ID3C, we normalize it
+        # anyway.
+        "netid": f"{environ['uid']}@uw.edu",
```
Comment (joverlee521) [inline lib/husky_musher/app.py:172]: This has changed in REDCap to just the NetID.

Comment (kairstenfay) [reply lib/husky_musher/app.py]: Good catch! Addressed in a new commit.

File: lib/husky_musher/app.py [hunk 2]
```diff
@@ -152,3 +156,74 @@ def main():
...
+    """
+    raw_affilations = environ.get("unscoped-affiliation", "")
+
+    # "Member" is uninteresting and uninformative; a generic catch-all.
+    # The empty string might arise from our fallback above.
+    affiliations = set(raw_affilations.split(";")) - {"member",""}
+
+    rules = [
+        ("faculty"  in affiliations,    {"affiliation": "faculty",  "affiliation_other": ""}),
+        ("student"  in affiliations,    {"affiliation": "student",  "affiliation_other": ""}),
```
Comment (joverlee521) [inline lib/husky_musher/app.py:223]: In REDCap, the "student" affiliation takes precedence (Are you a student, faculty, staff, or volunteer at the University of Washington? If you are both working and enrolled in classes at the University of Washington, please select "Student".) 

If I'm understanding this function correctly, should the `"student" in affiliations` condition be the first in the list?

Comment (kairstenfay) [reply lib/husky_musher/app.py:225]: Thanks! Fixed in a new commit.

**Review by joverlee521 — APPROVED** [review]

**Pull request merged**